packages/api/src/Question/Question.entity.ts

```diff
@@ -0,0 +1,41 @@
+import { ObjectType, Field, InputType, registerEnumType } from '@nestjs/graphql';
+
+export enum QuestionType {
+  codeComplete = 'codeComplete',
+  memoryCards = 'memoryCards'
+}
+
+registerEnumType(QuestionType, { name: 'QuestionType' });
+
+@ObjectType()
+export class CodeCompleteQuestion {
+  @Field()
+  file: string;
+
+  @Field(() => [String])
+  answers: string[];
+}
+
+@ObjectType()
+export class Question {
+  @Field()
+  type: QuestionType;
+
+  @Field(() => [CodeCompleteQuestion], { nullable: true })
+  files?: CodeCompleteQuestion[];
+
+  @Field(() => [[String]], { nullable: true })
+  pairs?: string[][];
+}
+
+@InputType()
+export class QuestionInput {
+  @Field()
+  pathId: string;
+
+  @Field()
+  moduleIndex: number;
+
+  @Field()
+  type?: string;
+}
```

Comment:
This would be a GREAT use case for a `Union` type. Let me know if you want some help setting this up.